--- a/kernel/nvidia-uvm/uvm_full_fault_handler.c 2016-03-17 03:59:59.000000000 +0100
+++ b/kernel/nvidia-uvm/uvm_full_fault_handler.c 2016-03-22 14:52:58.847247949 +0100
@@ -3095,7 +3095,7 @@
         pg = pfn_to_page(cpuPfn);
         UVM_PANIC_ON(!pg);
 
-        ret = VM_FAULT_MINOR;
+        ret = 0;
     }
 
     // Update the gpu state for the given cpu fault only if we have some GPU

--- a/kernel/nvidia-uvm/uvm_lite.c 2016-03-17 03:59:59.000000000 +0100
+++ b/kernel/nvidia-uvm/uvm_lite.c 2016-03-22 14:52:58.847247949 +0100
@@ -1283,5 +1283,5 @@
         // If we already have the page, then we must have earlier copied in the
         // data from the GPU. Therefore, avoid migrating.
         //
-        retValue = VM_FAULT_MINOR;
+        retValue = 0;
     }
